gilbreth_perception/src/recognition_node.cpp

```diff
@@ -0,0 +1,295 @@
+#include <ros/ros.h>
+#include <sensor_msgs/PointCloud2.h>
+#include <pcl_conversions/pcl_conversions.h>
+#include <pcl/point_types.h>
+#include <pcl/segmentation/sac_segmentation.h>
+#include <pcl/segmentation/extract_clusters.h>
+#include <pcl/io/pcd_io.h>
+#include <pcl/point_cloud.h>
+#include <pcl/correspondence.h>
+#include <pcl/features/normal_3d_omp.h>
+#include <pcl/features/shot_omp.h>
+#include <pcl/features/board.h>
+#include <pcl/features/normal_3d.h>
+#include <pcl/features/fpfh.h>
+#include <pcl/keypoints/uniform_sampling.h>
+#include <pcl/visualization/pcl_visualizer.h>
+#include <pcl/kdtree/kdtree_flann.h>
+#include <pcl/kdtree/impl/kdtree_flann.hpp>
+#include <pcl/common/transforms.h>
+#include <pcl/console/parse.h>
+#include <pcl/sample_consensus/method_types.h>
+#include <pcl/sample_consensus/model_types.h>
+#include <pcl/ModelCoefficients.h>
+#include <pcl/filters/extract_indices.h>
+#include <pcl/filters/passthrough.h>
+#include <pcl/filters/voxel_grid.h>
+#include <pcl/registration/ia_ransac.h>
+#include <tf/transform_listener.h>
+#include <geometry_msgs/PointStamped.h>
+#include <iostream>
+#include <cstdio>
+#include <fstream>
+#include <gilbreth_perception/ObjectDetection.h>
+
+typedef pcl::PointXYZ PointType;
+typedef pcl::Normal NormalType;
+
+//Algorithm params
+float descr_rad_ (0.02f);
+float down_sample_(0.01);
+float min_sample_distance_ (0.025f);
+float max_correspondence_distance_ (0.01f*0.01f);
+int nr_iterations_ (20);
+
+
+pcl::visualization::PCLVisualizer viewer ("recognition_result");
+ros::Publisher pub_tf, pub;
+
+
+struct Result
+{
+	std::string item_name;
+	int item_id;
+	float fitness_score;
+	pcl::PointXYZ pick_point;
+	Eigen::Matrix4f final_transformation;
+	EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+};
+
+
+
+void
+parseCommandLine (int argc, char *argv[])
+{
+	//General parameters
+	pcl::console::parse_argument (argc, argv, "--descr_rad", descr_rad_);
+	pcl::console::parse_argument (argc, argv, "--down_sample", down_sample_);
+	pcl::console::parse_argument (argc, argv, "--min_sample_distance",  min_sample_distance_);
+	pcl::console::parse_argument (argc, argv, "--max_correspondence_distance", max_correspondence_distance_);
+	pcl::console::parse_argument (argc, argv, "--nr_iterations", nr_iterations_);
+}
+
+
+void
+cloud_cb (const sensor_msgs::PointCloud2ConstPtr& cloud_msg,const tf::TransformListener& listener,int argc, char** argv)
+{
+
+	parseCommandLine (argc, argv);
+
+	// Load model settings
+	XmlRpc::XmlRpcValue model_map;
+	std::map<std::string,float> pick_orientation;
+	std::string package_path;
+	ros::NodeHandle ph("~");
+	ph.getParam("part_list",model_map);
```

Comment:
Why not load all the parameters using `getParam` instead of using `pcl::console::parse_argument`?